bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookieWriteLedgerTest.java

```diff
@@ -1294,6 +1300,91 @@ public void testLedgerCreateAdvSyncAsyncAddDuplicateEntryIds() throws Exception
         lh.close();
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testLedgerCreateAdvByteBufRefCnt() throws Exception {
+        long ledgerId = rng.nextLong();
+        ledgerId &= Long.MAX_VALUE;
+        if (!baseConf.getLedgerManagerFactoryClass().equals(LongHierarchicalLedgerManagerFactory.class)) {
+            // since LongHierarchicalLedgerManager supports ledgerIds of
+            // decimal length upto 19 digits but other
+            // LedgerManagers only upto 10 decimals
+            ledgerId %= 9999999999L;
+        }
+
+        final LedgerHandle lh = bkc.createLedgerAdv(ledgerId, 5, 3, 2, digestType, ledgerPassword, null);
+
+        final List<AbstractByteBufAllocator> allocs = Lists.newArrayList(
+                new PooledByteBufAllocator(true),
+                new PooledByteBufAllocator(false),
+                new UnpooledByteBufAllocator(true),
+                new UnpooledByteBufAllocator(false));
+
+        long entryId = 0;
+        for (AbstractByteBufAllocator alloc: allocs) {
+            final ByteBuf data = alloc.buffer(10);
+            data.writeBytes(("fragment0" + entryId).getBytes());
+            assertEquals("ref count on ByteBuf should be 1", 1, data.refCnt());
+
+            CompletableFuture<Integer> cf = new CompletableFuture<>();
+            lh.asyncAddEntry(entryId, data, (rc, handle, eId, qwcLatency, ctx) -> {
+                CompletableFuture<Integer> future = (CompletableFuture<Integer>) ctx;
+                future.complete(rc);
+            }, cf);
+
+            int rc = cf.get();
+            assertEquals("rc code is OK", BKException.Code.OK, rc);
+
+            TimeUnit.MILLISECONDS.sleep(250); // recycler runs asynchronously
```

Comment:
@dlg99 
this will be our next flaky test in the list.

during 4.5 development I had added some tests about ref counts but they turned to be quite flaky

Maybe it is better not to add assertions but only see that the asyncAddEntry completes with success.
I understand that this way we are not testing the whole change but I guess we will save us from debugging a new flaky test.